### ISSUE TITLE
[Request]: Tweets occupy full width of screen

### ISSUE THREAD

**Issue by freecicero** [GeopJr/Tuba#215]
### Describe the request

I like to run my mastodon feeds using the full width of the screen, like Whalebird can do.  Is this doable on Toable or something that could be added to the wish list?

### Implementation Details

- [X] This should be an option in settings.
- [ ] This should be only available to some fediverse backends. (Include which ones on the above field).
- [ ] This is client-only (and shouldn't sync with the instance).
- [X] This follows the [GNOME HIG](https://developer.gnome.org/hig/).

**Comment by nekohayo** [comment]
Isn't this the same as #118 ?

**Comment by freecicero** [comment]
Not as I read it I don't think so.... I like to use the full screen and large text to keep a rolling list of lastest toots on a big screen TV, so I don't want multicollumn, I want a single column all the way across the screen so that I can see as many toots as possible at one time. Currently the program seems to have a max width and posts do not keep extending laterally but stop. Thanks for consideration.

**Comment by GeopJr** [comment]
I don't think a "full width" UI really matches GNOME's design. Even Fractal (devel) (a Matrix client for GNOME) has some sort of column-like layout.

At the same time, it would need many changes to make it look somewhat good:

![image](https://user-images.githubusercontent.com/18014039/234368754-6c022cf9-f395-48e7-a209-f91a827b57d8.png)

![image](https://user-images.githubusercontent.com/18014039/234370065-e7ff0553-1ae2-495a-8e8a-644d8b186329.png)

![image](https://user-images.githubusercontent.com/18014039/234370117-269a9c66-8765-4b69-9d4b-d1a46da5f052.png)

![image](https://user-images.githubusercontent.com/18014039/234370163-3df3cbf1-e180-41e2-8e3d-5a3e3ca67a5c.png)

Limiting the width of everything but the content doesn't make much sense either, it will be the exact same as the column being aligned to the left side of the screen but without text wrapping.

**Comment by nekohayo** [comment]
Apparently this is what Whalebird looks like with long line lengths:

![whalebird desu](https://whalebird.social/assets/desktop/screenshot/tag-7d0d39002c04a7429401ea62913ad5b43047c9684911548df1e7c0ecdcade8eb.png)

So roughly twice the stretchable width? But its button widgets are all left-aligned, not equally-spaced-to-fill-the-width.

Beyond that kind of length, I'm not sure there is a way to do this without impairing typographic principles of legibility, see [this article](https://en.wikipedia.org/wiki/Line_length) ; that's mostly the reason why pretty much everything out there (apps, documents, websites, etc.) typically sets a maximum-width for paragraphs, or uses columns to cram more stuff horizontally: long lines generally make it harder to read.

As far as I can tell, it seems Tuba already wraps at somewhere around 80-90 characters (depending on your global desktop font scaling settings) on my computer.

That said,
* if it didn't cap the width, technically, users could always size down the width of the app to match whatever they want...
* otherwise, if GTK/desktop's font scaling is not `1.00` (in `gnome-tweaks`), I think it would make sense for Tuba to detect that and multiply the maximum width by the value of font scaling, so that it does not squeeze the text that is bigger than normal. This is an issue I noticed just now.

**Comment by freecicero** [comment]
Yes understandable and thanks for comments.  The major use case I would see is combining longer widths with using larger font sizes to be visible at a distance from the screen (or for those who might be sight impaired who want to take advantage of screen real estate and need the larger font sizes for that reason).
